pkg/config/parameters.go

```diff
@@ -499,6 +502,37 @@ func (p *ListenerParameters) Validate() error {
 		return fmt.Errorf("invalid per connections buffer limit bytes value %q set on listener, minimum value is 1", *p.PerConnectionBufferLimitBytes)
 	}
 
+	return p.SocketOptions.Validate()
+}
+
+// SocketOptions defines configurable socket options for Envoy listeners.
+type SocketOptions struct {
```

Comment:
(w/o knowing the details of TOS vs. Traffic Class) is it valid to auto detect the Listener ip family to be able to collapse these two into one field?

Reply:
Reliably auto-detecting could be challenging when listener address is set as `::`. We use [`Ipv4Compat: true`](https://github.com/projectcontour/contour/blob/984e42b99601f2a413708795f87ba780ba5f1a68/internal/envoy/v3/listener.go#L670) so Envoy accepts also IPv4 clients - unless IPv4 is disabled on the host where Envoy is running, I think.

Reply:
ah fair point, if we wanted to be clever could we maybe do something like "if listen address == :: then set the ipv4 and ipv6 options, otherwise set the appropriate option for the address family"?

Reply:
if in practice users may want to set different values between ipv4 and ipv6 seems more correct to keep these options/configurable values separate